oauth_client.go

```diff
@@ -125,3 +126,40 @@ func (s *oAuthClients) Create(ctx context.Context, organization string, options
 
 	return oc, nil
 }
+
+type OAuthCLientDestroyOptions struct {
+	CLIENT_ID    *string `jsonapi:"primary,oauth-client"`
+	Organization *string `jsonapi:"primary,organization"`
+}
+
+func (o OAuthCLientDestroyOptions) valid() error {
+	if !validString(o.CLIENT_ID) {
+		return errors.New("CLIENT_ID required")
+	}
+	return nil
+}
+
+func (s *oAuthClients) Delete(ctx context.Context, options OAuthCLientDestroyOptions) (*OAuthClient, error) {
+	if err := options.valid(); err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("oauth-clients/%s", url.QueryEscape(*options.CLIENT_ID))
```

Comment:
According to the [docs](https://www.terraform.io/docs/enterprise/api/oauth-clients.html#destroy-an-oauth-client), this URL should look like this: `organizations/:organization_name/oauth-clients/:id`